magma/digital.py

```diff
@@ -194,28 +283,24 @@ def getgpio(self):
     def unused(self):
         if self.is_input() or self.is_inout():
             raise TypeError("unused cannot be used with input/inout")
-        m.wire(self, DefineUnused()().I)
+        m.wire(m.bit(self), DefineUnused()().I)
 
     def undriven(self):
         if self.is_output() or self.is_inout():
             raise TypeError("undriven cannot be used with output/inout")
-        m.wire(DefineUndriven()().O, self)
-
-    def as_bits(self):
-        result = m.Bits[1]()
-        result[0] @= self
-        return result
-
-    @classmethod
-    def from_bits(cls, other):
-        if not isinstance(other, m.Bits) and not len(other) == 1:
-            raise TypeError("Can only convert from Bits[1] to Bit")
-        return other[0]
+        m.wire(DefineUndriven()().O, m.bit(self))
```

Comment:
Perhaps if we made `DefineUndriven` and `DefineUnused` parametrized by `type`, we could remove this circular dependency (on `m.bit`), also we could use `@=` to avoid `m.wire`?

Reply:
Agreed. Will work on this in the next batch of commits.